lib/linguist/generated.rb

```diff
@@ -698,13 +698,13 @@ def generated_graphql_relay?
     # Internal: Is this a generated Game Maker Studio (2) metadata file?
     #
     # All Game Maker Studio 2 generated files will be JSON, .yy or .yyp, and have
-    # a part that looks like "modelName: GMname" on the 3rd line
+    # a line that looks like "modelName: GMname" or "resourceType: GMname"
     #
     # Return true or false
     def generated_gamemakerstudio?
       return false unless ['.yy', '.yyp'].include? extname
       return false unless lines.count > 3
-      return lines[2].match(/\"modelName\"\:\s*\"GM/) ||
+      return lines.any? { |line| line.match(/\"(modelName|resourceType)\"\:\s*\"GM/) } ||
```

Comment:
Can we please keep this limited to a number of lines to protect against a massive rogue file crippling things.

Reply:
Agreed - this previous one on a fixed line worked perfect for the older GM files. Lets keep that line in there. Maybe add an extra line that scans just a single fixed line (like I mentioned) if that works.

Reply:
I would opt for keeping the lines lower (3 lines per file max) and just extend the amount of checks;
`$GM` as the first 3 letters of the first key
`resourceType` key with value starting with `GM`
`isDnD` in the key

things like that :thinking:

Reply:
>  just extend the amount of checks

@RobQuistNL I personally think having a check that looks at the first 200 lines is OK since the regex used is really simple and should still perform well. However, if you have any example of a GMStudio generated .yy file that doesn't match with the current rule, but matches one of your suggested checks, I'd be happy to make changes to fix that.